### PR TITLE
Avoid many instances of dynamic allocation

### PR DESCRIPTION
**Pull request by ALSchwalm**
This patchset make some significant changes to the allocation layout
of Mythril. Now, instead of heap allocation of devices and virtual
machines, these are allocated in the following way:

There is a static 'VirtualMachineSet' object that contains a list
of virtual machines and associated message passing contexts. This
structure is how inter-core and inter-vm communication occurs. Each
'VirtualMachine' now contains an ArrayVec of 'DynamicVirtualDevice'.

A 'DynamicVirtualDevice' is a wrapper around various EmulatedDevice
types that are _not_ part of the required guest architecture. For
example, a given PCI device is 'dynamic' in the sense that we cannot
state that it is part of the guest at compile time.

Non-dynamic EmulatedDevices are now stored explicitly as part of the
'StaticVirtualDevices' field of the 'VirtualMachine'. This makes it
significantly easier to reference a particular part of the guest
architecture (rather than having to look up a device by Port, for
example).

I also removed the stub EmulatedDevices we had and just started
ignoring the portio operations that are mapped to devices we don't
emulate. I believe this is the right move, architecturally. It may _not_
be the correct thing to do for memory mapped devices, but they
should trigger an EPT violation anyway.

### CODE REVIEW

File: mythril/src/vm.rs [hunk 1]
```diff
@@ -466,13 +528,13 @@ impl VirtualMachine {
...
-            .config
-            .virtual_devices()
-            .find_device(ident)
-            .ok_or_else(|| {
-                Error::MissingDevice("Unable to dispatch event".into())
-            })?;
+        let dev = match self.virtual_device_map.find_device(ident) {
+            Some(dev) => dev,
+            None => {
+                // warn!("Unable to dispatch event");
```
Comment (dlrobertson) [inline mythril/src/vm.rs:534]: Could this be changed to a TODO? The warn could also be useful

**Review by dlrobertson — COMMENTED** [review]
I actually really like the static device and dynamic device change... I think it helps simplify things a bit as we start developing more of the emulated devices

File: mythril/src/vm.rs (hunk 1 above)
Comment (ALSchwalm) [reply mythril/src/vm.rs]: Made this back into an error for non-portio devices

File: mythril/src/vm.rs [hunk 2]
```diff
@@ -1,10 +1,9 @@
-#![deny(missing_docs)]
+// #![deny(missing_docs)]
```
[the remark this answers is not in the document]
Comment (ALSchwalm) [reply mythril/src/vm.rs]: Done

**Pull request merged**